be/src/exec/vectorized/chunks_sorter_topn.cpp

```diff
@@ -392,27 +391,32 @@ Status ChunksSorterTopn::_hybrid_sort_common(RuntimeState* state, std::pair<Perm
         chunks.push_back(segment.chunk);
     }
 
+    // There are three parts of data
+    // part1: _merged_segment, the sorted one
+    // part2: BEFORE_LAST_RESULT, ∀ e ∈ part2, e < smallest(part1)
+    // part3: IN_LAST_RESULT, ∀ e ∈ part3, smallest(part1) <= e <= largest(part1)
+
+    // First, we find elements from part1
     if (first_size > 0) {
         big_chunk.reset(segments[new_permutation.first[0].chunk_index].chunk->clone_empty(first_size).release());
         append_by_permutation(big_chunk.get(), chunks, new_permutation.first);
         rows_to_keep -= first_size;
     }
 
-    // If rows_to_keep > 0, then second_size > 0 is always positive
-    // If rows_to_keep == 0 and sort type is rank, then we will process second part only if it's not empty
-    if (rows_to_keep > 0 || (_topn_type == TTopNType::RANK && second_size > 0)) {
-        // In case of rows_to_keep == 0, through _merged_segment.get_filter_array,
-        // all the rows that equal to the last value of merged segment are stored in the IN_LAST_RESULT,
-        // so we also need to consider this part in type rank
+    // Seoncd, there are two cases:
+    // case1: rows_to_keep == 0, which measn part1 itself suffice, we don't need to check part2 and part3
+    // case2: rows_to_keep > 0, which means part1 itself not suffice, we need to get more elements from part2 and part3,
+    // so we need to first merge part2 and part3 and then get elements
+    if (rows_to_keep > 0) {
```

Comment:
I think we could have a better name than rows_to_keep.

Reply:
Could you give one？